File: IOS/Sample Project/ObjC/Sample/Sample/ViewController.m
//
//  ViewController.m
//  Sample
//
//  Created by Hari Shankar on 28/06/17.
//  Copyright © 2017 Hari Shankar. All rights reserved.
//

#import "ViewController.h"
#import "Trova.h"
@interface ViewController (){
    Trova *trova;
}
@end

@implementation ViewController



- (void)viewDidLoad{
    [super viewDidLoad];
    
    trova = [[Trova alloc]init];
    
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle: nil];
    UIViewController *viewController = [storyboard instantiateViewControllerWithIdentifier:@"viewControllerIdentifier"];
    
    [trova updateTrovaCurrentViewController:self fromViewController:viewController];

    // Do any additional setup after loading the view, typically from a nib.
}

- (void)viewDidAppear:(BOOL)animated{
    [super viewDidAppear:animated];
    
}


- (IBAction)audioBtnTapped:(id)sender {
     [trova makeTrovaAudioCall:@"agentKey"];
}
- (IBAction)videoBtnTapped:(id)sender {
     [trova makeTrovaVideoCall:@"agentKey"];
}
- (IBAction)chatBtnTapped:(id)sender {
     [trova makeTrovaChat:@"agentKey"];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: IOS/Sample Project/Swift/Sample/Sample/ViewController.swift
//
//  ViewController.swift
//  Sample
//
//  Created by Hari Shankar on 21/06/17.
//  Copyright © 2017 Hari Shankar. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    let trova:Trova = Trova()
    var currentViewController:UIViewController?

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let storyBoard: UIStoryboard = UIStoryboard(name: "Main", bundle: nil)
        currentViewController = storyBoard.instantiateViewController(withIdentifier: "ViewControllerIdentifier") as! ViewController
        
        trova.updateCurrentViewController(self, from: currentViewController)

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    
    @IBAction func ChatButtonTapped(_ sender: Any) {
         trova.makeChat("agentKey")
    }

    @IBAction func VideoButtonTapped(_ sender: Any) {
         trova.makeVideoCall("agentKey")
    }
    
    @IBAction func AudioButtonTapped(_ sender: Any) {
         trova.makeAudioCall("agentKey")
    }
    
}

